api/src/utils/get-local-type.ts

```diff
@@ -119,7 +119,7 @@ export default function getLocalType(
 		if (special.includes('hash')) return 'hash';
 		if (special.includes('csv')) return 'csv';
 		if (special.includes('uuid')) return 'uuid';
-		if (type.startsWith('geometry')) {
+		if (type?.startsWith('geometry')) {
```

Comment:
Instead of this, what about iterate over `localTypeMap` and check if `column.data_type` starts with the `key` of `localTypeMap`. This way instead of returning `unknown` for `tinyint unsigned` it would return `integer`.
What are you thoughts @Oreilles @azrikahar?

Reply:
Not sure if that'll work in this particular use case, as I believe this type is the "local type", not the column data type 🤔 I might be mistaken though. @Oreilles ?

Reply:
Yeah I meant to do that procedure on line 113, so here `type` will be already `integer`.

Reply:
I'd say let's add `tinyint unsigned` to the localTypeMap instead!

Reply:
But then we will need to add every type variant to `localTypeMap` like `tinyint unsigned`, `tinyint zerofill`, `text collate ...`, etc. https://mariadb.com/kb/en/numeric-data-type-overview/
That's why I suggested to lookup only the first name.
Also I think this is not specific to geometry. The issue urged on that line because `type` was `undefined`.

Reply:
Seems error prone. Exhaustivity should be preferred IMO. As an example: type `time` whould return true if you check `"timestamp".startsWith('time')`

Reply:
I found another bug because of this line
https://github.com/directus/directus/blob/4fc02c7b6b7511346592cac4dbbf5b2b8445ad15/api/src/utils/get-local-type.ts#L113
Here we are splitting by ( and take first part, so types like
https://github.com/directus/directus/blob/4fc02c7b6b7511346592cac4dbbf5b2b8445ad15/api/src/utils/get-local-type.ts#L58
won't work and all tinyint's will be considered as boolean, it's incorrect for a lot of cases

Reply:
Saw this too, I'm on it

Reply:
@Oreilles mysql creates boolean type by default as tinyint(1), not as tinyint or tinyint(0), but in your solutions you map tinyint too. And I don't know, is it possible to use tinyint(0) as boolean?

Reply:
It probably is since that is how they were handled in previous versions.